instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/db/DbClientSpanNameExtractor.java

```diff
@@ -82,6 +82,9 @@ public String extract(REQUEST request) {
   private static final class SqlClientSpanNameExtractor<REQUEST>
       extends DbClientSpanNameExtractor<REQUEST> {
 
+    // a dedicated sanitizer just for extracting the operation and table name
+    private static final SqlStatementSanitizer sanitizer = SqlStatementSanitizer.create(true);
```

Comment:
This is probably the most controversial part of the PR: for the purpose of constructing a span name we need to call the query sanitize anyway, since it extracts the operation and table name.
It's all cached, so it shouldn't be that costly -- and I suspect that most users have sanitization enabled anyway, since it's the default; but in case we need to optimize for performance here I believe that we can introduce a new mode of processing to the sanitizer, that just stops after extracting the table (I'd rather complicate the sanitizer then introduce a builder for a span name extractor; one that duplicates a config setting to boot).